### PR TITLE
Fix-azure-and-cost

### PR DESCRIPTION
**Pull request by Mohannadcse**
Fixes the issues raised here:
- https://github.com/langroid/langroid/issues/234 
- https://github.com/langroid/langroid/issues/233 

It also updates the documentation and pytest to incorporate Azure env vars
Added a test case to check running a test case using openai after Azure doesn't fail

updated `test_token_usage.py` to check also based on Azure config.

There is no failure immediately after Azure test cases, but there are other failures 
image

### CODE REVIEW

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -24,6 +24,12 @@ jobs:
       GOOGLE_API_KEY: ${{ secrets.GOOGLE_API_KEY }}
       GOOGLE_CSE_ID: ${{ secrets.GOOGLE_CSE_ID }}
 
+      # Azure OpenAI env vars
+      AZURE_API_KEY: ${{ secrets.AZURE_API_KEY }}
+      AZURE_OPENAI_API_BASE: ${{ secrets.AZURE_OPENAI_API_BASE }}
+      AZURE_OPENAI_API_VERSION: "2023-05-15"
```
Comment (Mohannadcse) [inline .github/workflows/pytest.yml:30]: @pchalasani not quite sure how you want to make it, but left the value here.

**Comment by Mohannadcse** [comment]
After finishing running all test cases. Seems now that these failures aren't due to Azure config.

```
FAILED tests/main/test_repo_chunking.py::test_repo_chunking - Exception: Failed to get repo eugeneyan/testing-ml after 5 attempts.
FAILED tests/main/test_repo_loader.py::test_repo_loader - Exception: Failed to get repo eugeneyan/testing-ml after 5 attempts.
FAILED tests/main/test_stateless_tool_messages.py::test_llm_tool_message[True-SquareTool-Start by asking me to square the number 9, using a tool/function-call-81] - AttributeError: 'NoneType' object has no attribute 'name'
FAILED tests/main/sql_chat/test_sql_chat_agent.py::test_sql_chat_agent_query[False-How many employees are in Sales?-1] - AssertionError: assert '1' in 'DO-NOT-KNOW'
ERROR tests/main/test_momento_cache.py::test_real_store_and_retrieve - ValueError: MOMENTO_AUTH_TOKEN not set in .env file
```

**Comment by pchalasani** [comment]
there seem to be some transient fails. Re-running the pytest actions seems to always fix it

**Comment by Mohannadcse** [comment]
> there seem to be some transient fails. Re-running the pytest actions seems to always fix it

Great, I noticed that locally as well

File: .github/workflows/pytest.yml (hunk 1 above)
Comment (pchalasani) [inline .github/workflows/pytest.yml:30]: what about the `AZURE_GPT_MODEL_NAME`

Comment (Mohannadcse) [reply .github/workflows/pytest.yml:30]: I missed this, let me know when I can add it

Comment (pchalasani) [reply .github/workflows/pytest.yml:30]: I will add it

**Review by pchalasani — APPROVED** [review]

**Pull request merged**